rules/klayout/drc/testing/run_sc_regression.py

```diff
@@ -156,34 +161,53 @@ def call_simulator(arg):
         file = iname[0].split('/')
         for runset in rule_deck_path:
             arg = f"klayout -b -r {runset} -rd input={path} -rd report={file[-1]}_{x}.lyrdb -rd thr={thrCount} -rd conn_drc=true"
-            runs.append(arg)
-            x += 1
-
-    # Run DRC
-    with concurrent.futures.ProcessPoolExecutor(max_workers=thrCount) as executor:
-        for run in runs:
-            executor.submit(call_simulator, run)
-
-    # Get results
-    for path in args["--path"]:
-        x = 0
-        iname = path.split('.gds')
-        file = iname[0].split('/')
-        if "sc" in file[-1]:
-            continue
-        if "/" not in path:
-            path = f"sc/{path}"
-        iname = path.split('.gds')
-        file = iname[0].split('/')
-        for runset in rule_deck_path:
-            if os.path.exists(f"{iname[0]}_{x}.lyrdb"):
-                file, rule_deck, violations, status = get_results(runset, iname, file, x)
-                report.append([file, rule_deck, violations, status])
+            run_id = f"{runset}|{path}|{file[-1]}_{x}.lyrdb"
+            runs[run_id] = arg
             x += 1
 
-    with open(f'sc_drc_report.csv', 'w') as f:
-        writer = csv.writer(f, delimiter=',')
-        writer.writerows(report)
+    print("## We will run klayout {} runs".format(len(runs)))
+
+    # Run All DRC runs.
+    report = []
+
+    with concurrent.futures.ThreadPoolExecutor(max_workers=thrCount) as executor:
```

Comment:
using a test runner should allow you to achieve concurrency for free, see https://pytest-xdist.readthedocs.io/en/latest/